src/database.ts

```diff
@@ -1227,19 +1238,32 @@ class Database extends GrpcServiceObject {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const self = this;
     const callback =
-      typeof optionsOrCallback === 'function'
-        ? (optionsOrCallback as GetSessionsCallback)
-        : cb;
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb;
     const options =
       typeof optionsOrCallback === 'object'
-        ? (optionsOrCallback as GetSessionsOptions)
-        : {gaxOptions: {}};
-    const gaxOpts: CallOptions = options.gaxOptions as CallOptions;
-    const reqOpts = extend({}, options, {
+        ? optionsOrCallback
+        : ({} as GetSessionsOptions);
+    const gaxOpts = extend(true, {}, options.gaxOptions);
+    let reqOpts = extend({}, options, {
       database: this.formattedName_,
     });
-
     delete reqOpts.gaxOptions;
+
+    // Copy over pageSize and pageToken values from gaxOptions.
```

Comment:
Since we're in a breaking change, let's not even worry about supporting this improper usage. Most people have probably been using it in `reqOpts` already anyway.

Reply:
I completely agree with this but I think we have no choice but to do this while `CallOptions` still has `pageSize` and `pageToken` parameters (see my reply [here](https://github.com/googleapis/nodejs-spanner/pull/925/files#r416565445)). Someone is bound to set it here instead of in `reqOpts`.

Reply:
That's a good point! I sent a PR to gax to remove those options. After that, we can re-generate the docs, and remove them. I would be surprised if we had any users putting those options in there. Already, turning off auto-pagination and setting `pageSize` and `pageToken` manually is rare. Setting options through gax (a very unfamiliar concept for a user) would be even more unlikely.

I would be okay with still dropping our backup support while we wait for the changes from the linked PR to take effect. Otherwise, I think waiting is worth it, to avoid the complexity of support. Of course, we can always come back and remove the code later, so whatever everyone thinks is best :)

Reply:
@stephenplusplus do you know how long it would take to get the gax PR merged and into a release?

The main concern I have is that this PR needs to go into v5.0.0. We need v5.0.0 to go out asap since it's got the backup & restore support which is already a GA'ed feature but is currently not supported by a Node.js release.

So if we think the gax change will take more than a week to get through, then we should merge this as is then remove the code later.

Reply:
I'm not sure how long that would take. Hopefully @alexander-fenster can take a look at that PR (https://github.com/googleapis/gax-nodejs/pull/821) when he's available and give us a better idea.

Reply:
We just made a semver major for gax a few weeks ago. From one hand, major releases are cheap, but from another hand, having too many majors makes it harder to hot fix security issues so I would better wait for some time before making gax v3.0.0. So I would probably suggest you go to v5.0.0 here as is. What do you think @skuruppu @stephenplusplus?

Reply:
Thanks @alexander-fenster. It's okay with me to wait. The temporary code won't be too bad. Alternatively, I still think we're safe to remove support for the options anyway, and if anybody in the world notices, we can release a hotfix. But that's completely your call @skuruppu.

Reply:
I agree that it's so very unlikely that anyone would be trying to set these options through CallOptions. But just in case, we'll leave the temporary code. It won't be too hard to remove it when gax v3.0.0 is ready. I'm open to doing another major release whenever we need to.